pkg/transport/generic_transport_controller.go

```diff
@@ -410,44 +506,241 @@ func (c *genericTransportController) getObjectsFromWDS(ctx context.Context, bind
+	} else if !errors.IsNotFound(err) {
+		c.logger.Error(err, "Inconceivable failure to fetch parameter ConfigMap", "dest", dest, "forBinding", bindingName)
+	}
+	return props
+}
+
+func enumerateParmsInConfigMap(parmCfgMap *corev1.ConfigMap) func(yield func(key, val string) bool) {
+	return func(yield func(key, val string) bool) {
+		if parmCfgMap == nil {
+			return
+		}
+		for key, val := range parmCfgMap.Data {
+			if token.IsIdentifier(key) && !yield(key, val) {
+				return
+			}
+		}
+		for key, val := range parmCfgMap.BinaryData {
+			if token.IsIdentifier(key) && !yield(key, string(val)) {
+				return
+			}
+		}
+	}
 }
 
-func (c *genericTransportController) propagateWrappedObjectToClusters(ctx context.Context, desiredWrappedObject *unstructured.Unstructured,
+func (c *genericTransportController) setBindingSensitivities(bindingName string, dests sets.Set[v1alpha1.Destination]) {
+	c.Lock()
+	defer c.Unlock()
+	if dests == nil {
+		delete(c.bindingSensitiveDestinations, bindingName)
+	} else {
+		c.bindingSensitiveDestinations[bindingName] = dests
+	}
+}
+
+func (c *genericTransportController) handleParmCfgMapEvent(cmName string, parmCfgMap *corev1.ConfigMap, event string) {
+	dest := v1alpha1.Destination{ClusterId: cmName}
+	c.RLock()
+	defer c.RUnlock()
+	sameProps := equalProperties(cmName, c.destinationProperties[dest], parmCfgMap)
+	if parmCfgMap != nil {
+		if sameProps {
+			return
+		}
+	} else {
+		if _, has := c.destinationProperties[dest]; !has {
+			return
+		}
+	}
+	delete(c.destinationProperties, dest)
+	if sameProps {
+		return
+	}
+	for bindingName, dests := range c.bindingSensitiveDestinations {
+		if dests.Has(dest) {
+			c.logger.V(5).Info("Enqueuing reference to Binding that depends on destination properties", "binding", bindingName, "destination", dest, "event", event)
+			c.workqueue.Add(bindingName)
+		}
+	}
+}
+
+// equalProperties reveals whether the new version of the parameter ConfigMap implies the same old customization parameters
+func equalProperties(destName string, oldProps map[string]string, newConfigMap *corev1.ConfigMap) bool {
+	var newNumProperties int = 1
+	visitedClusterName := false
+	mismatch := false
+	enumerateParmsInConfigMap(newConfigMap)(func(key, newVal string) bool {
+		if key == "clusterName" {
+			visitedClusterName = true
+		} else {
+			newNumProperties++
+		}
+		if oldVal, had := oldProps[key]; !(had && oldVal == newVal) {
+			mismatch = true
+			return false
+		}
+		return true
+	})
+	if !visitedClusterName && oldProps["clusterName"] != destName {
+		return false
+	}
+	return newNumProperties == len(oldProps) && !mismatch
+}
+
+// customizeForDest customizes the given object for the given destination,
+// if any customization is called for. The returned boolean indicates whether
+// any customization was called for.
+func (c *genericTransportController) customizeForDest(object *unstructured.Unstructured, dest string, defs map[string]string) (*unstructured.Unstructured, []string, bool) {
```

Comment:
can you please use full names and not shortcuts? this is very confusing.
dest, defs..

Reply:
Made names longer.

I also made the terminology more consistent. I was previously using "properties" in some places, "parameters" in others. I have now tried to consistently use "properties".